### PR TITLE
Update 'GetBuildGraph' to understand VS2017 solutions

### LINKED ISSUE #463: Support MSBuild projects in build graph
The `GetBuildGraph()` function reads project.json to find dependencies. Dependencies for projects that have ported to csproj are not correctly identified, so the build graph builds these repos in the wrong order. We will need to invoke `/t:GenerateRestoreGraphFile` to inspect which packages are inside the solution and which packages they depend upon.

### PR DESCRIPTION
**Pull request by natemcmaster**
Updates the topological sort to find dependencies and packages from VS 2017 solutions.

TODO: Update KoreBuild dev branch to install the MSBuild variant of dotnet-cli.

Resolves #463

### CODE REVIEW

File: makefile.shade [hunk 1]
```diff
@@ -802,12 +812,138 @@ functions
...
+            }
+
+            private static void ResolveMsbuildProjects(RepositoryInfo info, string repo)
+            {
+                foreach (var sln in Directory.EnumerateFiles(repo, "*.sln"))
+                {
+                    var lines = File.ReadAllLines(sln);
+                    if (lines.Count() < 2) continue;
+                    var versionLine = lines[1];
+                    if (versionLine != "# Visual Studio 15") continue;
```
Comment (BrennanConroy) [inline makefile.shade]: Not sure if we want to future proof this and check for >=15

File: makefile.shade [hunk 2]
```diff
@@ -802,12 +812,138 @@ functions
...
+            public static RepositoryInfo Create(string repo)
             {
                 var info = new RepositoryInfo { Name = repo };
+                ResolveProjectJsonProjects(info, repo);
+                ResolveMsbuildProjects(info, repo);
+                info.DependencyNames.ExceptWith(info.RepositoryNames);
+                return info;
+            }
+
+            private static void ResolveMsbuildProjects(RepositoryInfo info, string repo)
```
Comment (BrennanConroy) [inline makefile.shade]: Make build capital like the other function

**Comment by natemcmaster** [comment]
:up: :date:

**Comment by BrennanConroy** [comment]
Not very familiar with this area, but looks alright to me.

**Comment by natemcmaster** [comment]
@Eilon can you think of anyone else that would be okay to sign off while Pranav is out?

File: makefile.shade [hunk 3]
```diff
@@ -802,12 +812,148 @@ functions
...
+                        var match = Regex.Match(line, @"^# Visual Studio (\d+)\s?");
+                        if (match == null || match.Groups.Count < 2)
+                        {
+                            continue;
+                        }
+                        int version;
+                        if (int.TryParse(match.Groups[1].Value, out version) && version >= 15)
+                        {
+                            ResolveMsBuildProject(info, sln);
+                            return;
```
Comment (Eilon) [inline makefile.shade]: This stops after the first project it finds? Or maybe I'm not understanding this?

**Comment by natemcmaster** [comment]
Btw, for anyone trying to verify, this is the batching that Universe should use:
```
Building repositories in batches: 
1 - PlatformAbstractions
2 - Common
3 - Testing
4 - JsonPatch, FileSystem, DependencyInjection, Microsoft.Data.Sqlite
5 - Configuration, EventNotification
6 - Options, Logging
7 - DotNetTools, HtmlAbstractions, HttpAbstractions, Caching
8 - Razor
9 - RazorTooling, Hosting
10 - DataProtection, EntityFramework, WebListener, KestrelHttpServer, BrowserLink
11 - EntityFramework.Tools, BasicMiddleware, Scaffolding
12 - IISIntegration
13 - AzureIntegration, ServerTests, ResponseCaching, Session, CORS, Routing, StaticFiles, Proxy
14 - Diagnostics, Security, Antiforgery, Localization
15 - WebSockets, Mvc, MetaPackages
16 - MvcPrecompilation, Identity
17 - SignalR, Entropy, MusicStore
```
You can inspect these changes work by checking out the feature/msbuild branch of DotNetTools and running `build show-build-graph`

**Comment by Eilon** [comment]
@natemcmaster I think between you, @BrennanConroy , and perhaps @anurse , we should be good. And ultimately, either it works, or it doesn't 😄

File: makefile.shade (hunk 3 above)
Comment (natemcmaster) [inline makefile.shade]: aha, good catch. Yes, this should have been 'break'

Comment (Eilon) [inline makefile.shade]: I knew I wasn't crazy... that's what the voices in my head were telling me! 😄

**Pull request merged**